src/libsystemd-network/sd-dhcp6-lease.c

```diff
@@ -809,9 +890,15 @@ static int dhcp6_lease_parse_message(
 }
 
 static sd_dhcp6_lease *dhcp6_lease_free(sd_dhcp6_lease *lease) {
+        sd_dhcp6_option *dhcp6_option;
+
         if (!lease)
                 return NULL;
 
+        SET_FOREACH(dhcp6_option, lease->vendor_options)
+                dhcp6_vendor_option_free(dhcp6_option);
```

Comment:
By using `dhcp6_option_hash_ops`, this loop can be dropped.

Reply:
Fixed